### PR TITLE
feat(perf): Open Performance landing page widgets in Discover

### PR DESCRIPTION
**Pull request by gggritso**
Adds an "Open in Discover" button to some widgets on the performance landing page. Not all of them can be nicely represented in Discover (histograms, vitals), so this is enabled for all _area_ graph widgets.

**If the widget supports "Open in Discover":**

Screen Shot 2022-07-15 at 5 15 08 PM

**If the widget does not support "Open in Discover":**
Screen Shot 2022-07-15 at 5 13 45 PM

**Video:**

https://user-images.githubusercontent.com/989898/179313003-d091fa62-2eb5-4955-a6c1-f884a5ebbda9.mov

## Changes

- Replace `CompactSelect` with `CompositeSelect`
- Add "Open in Discover" button to widget meatball menu

Closes PERF-1576

### CODE REVIEW

**Comment by gggritso** [comment]
@k-fish could you take a look and let me know if the code makes sense so far?

File: static/app/views/performance/landing/widgets/components/widgetContainer.tsx [hunk 1]
```diff
@@ -97,13 +101,25 @@ const _WidgetContainer = (props: Props) => {
...
+  // Construct an EventView that matches this widget's definition. The
+  // `eventView` from the props is the _landing page_ EventView, which is different
+  const widgetEventView = props.eventView.clone();
+  widgetEventView.name = chartDefinition.title;
+  widgetEventView.yAxis = chartDefinition.fields[0]; // All current widgets only have one field
+  widgetEventView.display = DisplayModes.PREVIOUS;
+  widgetEventView.fields = ['transaction', 'project', ...chartDefinition.fields].map(
+    fieldName => ({field: fieldName} as Field)
+  );
+
```
Comment (k-fish) [inline static/app/views/performance/landing/widgets/components/widgetContainer.tsx:114]: Might be nice to move this into its own function for clarity. `function getEventViewForActions` ?

File: static/app/views/performance/landing/widgets/components/widgetContainer.tsx [hunk 2]
```diff
@@ -173,18 +195,55 @@ export const WidgetContainerActions = ({
     );
   }
 
+  function handleWidgetActionChange(value) {
+    if (value === 'open_in_discover') {
+      window.location.href =
```
Comment (k-fish) [inline static/app/views/performance/landing/widgets/components/widgetContainer.tsx:200]: For consistency, we usually use `browserHistory.push`.

**Review by k-fish — APPROVED** [review]
Looks great @gggritso 👏 

I think this will work quite well, looking at the ticket now that I'd had a minute, we aren't touching any of the widgets that modify eventView (none of the 2-per-row or 'list action' widgets) so this should do the trick for now.

Even if the ticket doesn't cover all of them, I'd look in https://github.com/getsentry/sentry/tree/master/static/app/views/performance/landing/views and cover all the smaller top charts for each view (eg. https://github.com/getsentry/sentry/blob/master/static/app/views/performance/landing/views/allTransactionsView.tsx#L17-L26)  since I think this should work for all of them, would be nice for consistency.

File: static/app/views/performance/landing/widgets/components/widgetContainer.tsx (hunk 1 above)
Comment (gggritso) [reply static/app/views/performance/landing/widgets/components/widgetContainer.tsx]: Done in [28cf24b](https://github.com/getsentry/sentry/pull/36745/commits/28cf24b65d5704c1db8219ed0304f7dd602b03aa) 👌🏻

**Comment by k-fish** [comment]
@gggritso sounds good to me 👍

**Review by udameli — APPROVED** [review]
looks good!

**Pull request merged**